File: app/src/main/java/hk/collaction/freehkkai/ui/activity/MainActivity.java
package hk.collaction.freehkkai.ui.activity;

import android.os.Bundle;
import android.os.Handler;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.widget.Toolbar;
import android.widget.RelativeLayout;
import android.widget.Toast;

import com.google.android.gms.ads.AdView;

import butterknife.BindView;
import butterknife.ButterKnife;
import hk.collaction.freehkkai.C;
import hk.collaction.freehkkai.R;
import hk.collaction.freehkkai.ui.fragment.MainFragment;

public class MainActivity extends BaseActivity {

	private boolean doubleBackToExitPressedOnce;

	@BindView(R.id.toolbar)
	Toolbar toolbar;

	private AdView adView;

	@BindView(R.id.adLayout)
	RelativeLayout adLayout;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_container_adview);
		ButterKnife.bind(this);
		setSupportActionBar(toolbar);

		ActionBar ab = initActionBar(getSupportActionBar(), R.string.app_name);
		ab.setIcon(R.mipmap.ic_launcher);
		ab.setDisplayHomeAsUpEnabled(false);
		ab.setHomeButtonEnabled(false);

		adView = C.initAdView(mContext, adLayout);

		Fragment fragment = new MainFragment();
		FragmentManager fragmentManager = getSupportFragmentManager();
		fragmentManager.beginTransaction()
				.replace(R.id.container, fragment)
				.commit();
	}

	@Override
	public void onDestroy() {
		if (adView != null) {
			adView.removeAllViews();
			adView.destroy();
		}
		super.onDestroy();
	}

	@Override
	public void onBackPressed() {
		if (doubleBackToExitPressedOnce) {
			finish();
		} else {
			doubleBackToExitPressedOnce = true;
			Toast.makeText(mContext, "請按多次返回鍵以離開程式", Toast.LENGTH_SHORT).show();

			new Handler().postDelayed(new Runnable() {
				@Override
				public void run() {
					doubleBackToExitPressedOnce = false;
				}
			}, 2000);
		}
	}
}

File: app/src/main/java/hk/collaction/freehkkai/ui/activity/MainActivity.kt
package hk.collaction.freehkkai.ui.activity

import android.os.Handler
import android.widget.Toast
import androidx.fragment.app.Fragment
import hk.collaction.freehkkai.R
import hk.collaction.freehkkai.ui.fragment.MainFragment


class MainActivity : BaseFragmentActivity() {
    private var doubleBackToExitPressedOnce = false

    override var fragment: Fragment? = MainFragment()
    override var titleId: Int? = R.string.app_name


    override fun onBackPressed() {
        if (doubleBackToExitPressedOnce) {
            finish()
        } else {
            doubleBackToExitPressedOnce = true
            Toast.makeText(this, "請按多次返回鍵以離開程式", Toast.LENGTH_SHORT).show()
            Handler().postDelayed({ doubleBackToExitPressedOnce = false }, 2000)
        }
    }
}